lib/streamlit/script_request_queue.py

```diff
@@ -32,32 +30,31 @@ class ScriptRequest(Enum):
     SHUTDOWN = "SHUTDOWN"
 
 
-@attr.s(auto_attribs=True, slots=True)
-class RerunData:
-    """Data attached to RERUN requests."""
+class RerunData(NamedTuple):
```

Comment:
Instead of making it a named tuple, we should keep using attrs and use the `frozen` argument for it

Reply:
This is a matter of taste, and since you're the original author, I'll defer to yours here. But I'm not a big fan of `@attr.s` now that `@dataclass` is a thing:

- It adds a dependency to our codebase
- It adds cognitive overhead to our codebase (I really dislike their naming! `auto_attributes=True` and `slots=True` are weird property names that I always need to look up the meaning of. `attr.s` is cute to the point of obfuscation, IMO. Same with `attr.ib`.)
- It's non-standard. This is basically a hand-wave-y rehashing of the previous point, but `NamedTuple` is _the_ standard way of writing immutable Python types. Why would we use `@attr.s(auto_attribs=True, slots=True, frozen=True)` instead of subclassing from `NamedTuple`?

I'm much more sanguine about `@attr.s` vs `@dataclass` - I prefer dataclass because it's part of the standard lib and easier to use (IMO). But `@attr.s` vs `NamedTuple` feels more clear-cut in NamedTuple's favor.